ha/virsh/tests/resource_pgsql_test.sh

```diff
@@ -0,0 +1,175 @@
+# shellcheck shell=bash
+
+# Based on LP: #2013084; and
+# https://wiki.clusterlabs.org/wiki/PgSQL_Replicated_Cluster; and
+# https://clusterlabs.github.io/PAF
+
+# shellcheck disable=SC1090
+. "$(dirname "$0")/helper/test_helper.sh"
+# shellcheck disable=SC1090
+. "$(dirname "$0")/../vm_utils.sh"
+
+
+setup_pgsql() {
```

Comment:
I'd split this function into multiple ones to make it easier to understand, such as one to set up the cluster properly, another one to set up the master and another one to the replica.

Reply:
Thanks, Lucas! I split the functions where it made sense without changing the order of the calls in that single (big) function.